compiler/AST/astutil.cpp

```diff
@@ -856,12 +856,30 @@ bool isTypeExpr(Expr* expr) {
 
     } else if (FnSymbol* fn = call->resolvedFunction()) {
       retval = fn->retTag == RET_TYPE;
+
+    // TODO: Is it safe to just check if the base expression is a type?
+    } else if (isTypeConstructorWithRuntimeTypeActual(call)) {
+      retval = true;
```

Comment:
I am curious to know the answer to that question. Can be done outside this PR, though.